File: payroll-api/src/main/java/com/company/payroll/model/StaffResignation.java
package com.company.payroll.model;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonProperty;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
public class StaffResignation {
    private Integer resignId;
    private String reason;
    private LocalDate resignDate;
    @Schema(description="Resign status: 0=ongoing, 1=completed, 2=rejected")
    private Byte resignStatus;
    @JsonProperty("filename")
    private String fileName;
    @JsonProperty("size")
    private long fileSize;
//    private String attachment;
    private String filePath;
    private Integer staffId;
    private StaffDetails staffDetails;
//    private Integer eId;
//    private Integer mId;
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/model/StaffResignation.kt
package com.company.payroll.model

import com.company.payroll.annotation.NoArg
import lombok.AllArgsConstructor
import lombok.RequiredArgsConstructor
import java.time.LocalDate

@NoArg
data class StaffResignation(
    var resignId: String,
    var reason: String,
    var resignDate: LocalDate,
    var resignStatus: UByte,
    var fileName: String?,
    var fileSize: Long?,
    var filePath: String?,
    var staffId: Int,
    var staffDetails: StaffDetails?
)
